src/ui/TrackViewer.tsx

```diff
@@ -405,6 +413,38 @@ export class TrackViewer extends Object2D {
         this.layoutTrackRows(animate);
     }
 
+    setTrackIndex(track: Track, index: number, animate = true) {
+        // re-orders tracks in this.tracks[] and then calls this.layoutTrackRows(animate);
+
+        // apply array bounds check
+        index = Math.min(Math.max(index, 0), this.tracks.length - 1);
```

Comment:
Is it a good idea to mutate an argument?

Reply:
No! Updated.